pkg/controller/pod_driver.go

```diff
@@ -438,49 +450,42 @@ func (p *PodDriver) podDeletedHandler(ctx context.Context, pod *corev1.Pod) (Res
 		go p.CleanUpCache(context.TODO(), pod)
 		// stop fuse fd and clean up socket
 		go passfd.GlobalFds.StopFd(context.TODO(), hashVal)
+		// remove finalizer of pod
+		if err := resource.RemoveFinalizer(ctx, p.Client, pod, common.Finalizer); err != nil {
+			log.Error(err, "remove pod finalizer error")
+			return Result{}, err
+		}
 		return Result{}, nil
 	}
 
-	lock := config.GetPodLock(hashVal)
-	lock.Lock()
-	defer lock.Unlock()
-
 	// create
-	log.Info("pod targetPath not empty, need to create pod")
-
-	// check pod delete
-	_, err = p.Client.GetPod(ctx, pod.Name, pod.Namespace)
-	if err == nil || apierrors.IsNotFound(err) {
-		needCreate, err := p.needCreateMountPod(ctx, pod.Labels[common.PodUniqueIdLabelKey], hashVal)
-		if err != nil {
-			return Result{}, err
-		}
-		if needCreate {
-			// create pod
-			newPodName := podmount.GenPodNameByUniqueId(pod.Labels[common.PodUniqueIdLabelKey], true)
-			log.Info("need to create a new one", "newPodName", newPodName)
-			// delete tmp file
-			log.Info("delete tmp state file because it is not smoothly upgrade")
-			_ = util.DoWithTimeout(ctx, defaultCheckoutTimeout, func() error {
-				return os.Remove(path.Join("/tmp", hashVal, "state1.json"))
-			})
-			newPod, err := p.newMountPod(ctx, pod, newPodName)
-			if err == nil {
-				_, err = p.Client.CreatePod(ctx, newPod)
-				if err != nil {
-					log.Error(err, "Create pod")
-				}
+	if notAvailable {
+		// create pod
+		newPodName := podmount.GenPodNameByUniqueId(pod.Labels[common.PodUniqueIdLabelKey], true)
+		log.Info("pod targetPath not empty, need to create a new one", "newPodName", newPodName)
```

Comment:
why not available mount pods mean **pod targetPath not empty**?

Reply:
How to manually delete a pod in a delayed deletion state(with no refs)

Reply:
not available means there are no available mount pod in cluster. If pod targetPath not empty and no available , need to create a new one.

Reply:
If manually delete the delay deleted pod, it will not be handled as we discussed before.

Reply:
If I manually delete a delay pod, will a new one be created? even if there's no reference?

Reply:
> not available means there are no available mount pod in cluster. If pod targetPath not empty and no available , need to create a new one.

should we check `len(existTargets) > 0` here?

check `len(existTargets) == 0` before but with other condition..

https://github.com/juicedata/juicefs-csi-driver/blob/895ddad32d50e0325fb469aec1a6f97a12ed17e8/pkg/controller/pod_driver.go#L435